mrtt-ui/src/components/EcologicalStatusAndOutcomes/EcologicalOutcomesRow.js

```diff
@@ -0,0 +1,194 @@
+import { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import { Box, Checkbox, List, ListItem, MenuItem, TextField, Typography } from '@mui/material'
+
+import {
+  NestedLabel1,
+  NestedLabel3,
+  TabularSectionDiv,
+  TabularLabel,
+  TabularInputSection
+} from '../../styles/forms'
+import { comparisonOptions } from '../../data/ecologicalOptions'
+import { ErrorText } from '../../styles/typography'
+
+const EcologicalOutcomesRow = ({
+  index,
+  mainLabel,
+  secondaryLabel,
+  indicator,
+  metric,
+  measurement,
+  unit,
+  comparison,
+  measurementComparison,
+  linkedAims,
+  selectedAims,
+  updateItem
+}) => {
+  const [initialMeasurement, setInitialMeasurement] = useState('')
+  const [currentMeasurement, setCurrentMeasurement] = useState('')
+
+  const [initialUnit, setInitialUnit] = useState('')
+  const [currentUnit, setCurrentUnit] = useState('')
+
+  const [initialComparison, setInitialComparison] = useState('')
+  const [currentComparison, setCurrentComparison] = useState('')
+
+  const [initialMeasurementComparison, setInitialMeasurementComparison] = useState('')
+  const [currentMeasurementComparison, setCurrentMeasurementComparison] = useState('')
+
+  const [initialLinkedAims, setInitialLinkedAims] = useState([])
+  const [currentLinkedAims, setCurrentLinkedAims] = useState([])
+
+  const handleUpdate = () => {
+    if (currentMeasurement !== initialMeasurement) {
```

Comment:
what problem is the initial value vs current value solving?

Reply:
I'm thinking to scrap `handleUpdate` and just use `updateItem({ index, currentUnit })` onBlur

I wanted to avoid having the same prop names as the `useState` values. We are also passing previous form values to this component (whether they are empty or not). That is why I started the initial/current value part but i'm realizing that is kind of unnecessary. I still want to differentiate between the prop and the useState val however

Reply:
let me know if that is confusing...